modules/lib/types-dag.nix

```diff
@@ -65,7 +65,21 @@ rec {
     mkOptionType rec {
       name = "dagOf";
       description = "DAG of ${elemType.description}";
-      inherit (attrEquivalent) check merge emptyValue;
+      check = {
+        __functor = _self: attrEquivalent.check;
```

Comment:
It smells a bit off that this doesn't handle the case where `self.merge.v2` returns errors. Is it assumed that it is handled by `attrEquivialent.check`?

As is, this looks _effectively_ equivalent to just doing `inherit (attrEquivialent) check`, and assuming that v2 is handled there.

If you're fine with that assumption, maybe it's worth noting in a comment? But it _kinda_ feels like `check.isV2MergeCoherent = true` is a lie.